src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListEnumeratorTypeSymbol.cs

```diff
@@ -0,0 +1,265 @@
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedReadOnlyListEnumeratorTypeSymbol : NamedTypeSymbol
+    {
+        internal static NamedTypeSymbol Create(SourceModuleSymbol containingModule, string name)
+        {
+            return new SynthesizedReadOnlyListEnumeratorTypeSymbol(containingModule, name);
+        }
+
+        private readonly SourceModuleSymbol _containingModule;
+        private readonly ImmutableArray<NamedTypeSymbol> _interfaces;
+        private readonly ImmutableArray<Symbol> _members;
+        private readonly FieldSymbol _itemField;
+        private readonly FieldSymbol _moveNextCalledField;
+
+        private SynthesizedReadOnlyListEnumeratorTypeSymbol(SourceModuleSymbol containingModule, string name)
+        {
+            var compilation = containingModule.DeclaringCompilation;
+
+            _containingModule = containingModule;
+            Name = name;
+            var typeParameter = new SynthesizedReadOnlyListTypeParameterSymbol(this);
+            TypeParameters = ImmutableArray.Create<TypeParameterSymbol>(typeParameter);
+            var typeArgs = TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
+
+            _itemField = new SynthesizedFieldSymbol(this, typeParameter, "_item", isReadOnly: true);
+            _moveNextCalledField = new SynthesizedFieldSymbol(this, compilation.GetSpecialType(SpecialType.System_Boolean), "_moveNextCalled", isReadOnly: false);
+
+            var iDisposable = compilation.GetSpecialType(SpecialType.System_IDisposable);
+            var iEnumerator = compilation.GetSpecialType(SpecialType.System_Collections_IEnumerator);
+            var iEnumeratorT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerator_T).Construct(typeArgs);
+            _interfaces = ImmutableArray.Create(
+                iDisposable,
+                iEnumerator,
+                iEnumeratorT);
+
+            var membersBuilder = ArrayBuilder<Symbol>.GetInstance();
+            membersBuilder.Add(_itemField);
+            membersBuilder.Add(_moveNextCalledField);
+            membersBuilder.Add(
+                new SynthesizedReadOnlyListEnumeratorConstructor(this, typeParameter));
+            addProperty(membersBuilder,
+                new SynthesizedReadOnlyListProperty(
+                    this,
+                    (PropertySymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Collections_IEnumerator__Current),
+                    static (f, method, interfaceMethod) =>
+                    {
+                        var containingType = (SynthesizedReadOnlyListEnumeratorTypeSymbol)method.ContainingType;
+                        var itemField = containingType._itemField;
+                        var itemFieldReference = f.Field(f.This(), itemField);
+                        return f.Return(f.Convert(method.ReturnType, itemFieldReference));
```

Comment:
>return f.Return(f.Convert(method.ReturnType, itemFieldReference)); [](http://example.com/codeflow?start=24&length=66)

Consider including a comment with the generated code in each of these method bodies. For instance: `// return (object)_item;`